controller/two_lane_queue.go

```diff
@@ -91,39 +86,47 @@ func (tlq *twoLaneQueue) runConsumer() {
 	// Shutdown flags.
 	fast, slow := true, true
 	go func() {
+		// When both producer queues are shutdown stop the consumerQueue.
+		defer tlq.consumerQueue.ShutDown()
+		// While any of the queues is still running, try to read off of them.
 		for fast || slow {
 			// By default drain the fast lane.
+			// Channels in select are picked random, so first
+			// we have a select that only looks at the fast lane queue.
```

Comment:
my point about documenting things was that I don't think it's clear to folks that `default:` allows you to escape if no channels are ready to receive.